pkg/check/act/act.go

```diff
@@ -0,0 +1,179 @@
+	return Options{
+		FileName:      "act",
+		FileSize:      1 * 1024,
+		PostageAmount: 420000000,
+		PostageDepth:  17,
+		PostageLabel:  "act-label",
+		Seed:          0,
+	}
+}
+
+// compile check whether Check implements interface
+var _ beekeeper.Action = (*Check)(nil)
+
+// Check instance
+type Check struct {
+	logger logging.Logger
+}
+
+// NewCheck returns new check
+func NewCheck(logger logging.Logger) beekeeper.Action {
+	return &Check{
+		logger: logger,
+	}
+}
+
+// Run executes act check
+func (c *Check) Run(ctx context.Context, cluster orchestration.Cluster, opts interface{}) (err error) {
+	c.logger.Info("ACT test started")
+	o, ok := opts.(Options)
+	if !ok {
+		return fmt.Errorf("invalid options type")
+	}
+
+	clients, err := cluster.NodesClients(ctx)
+	if err != nil {
+		return err
+	}
+
+	sortedNodes := cluster.NodeNames()
+	upNodeName := sortedNodes[0]
+	upClient := clients[upNodeName]
+	addr, _ := upClient.Addresses(ctx)
+	publisher, _ := swarm.ParseHexAddress(addr.PublicKey)
+
+	rnds := random.PseudoGenerators(o.Seed, 1)
+
+	fileName := fmt.Sprintf("%s-%s-%d", o.FileName, upNodeName, rnds[0].Int())
+	postagelabel := fmt.Sprintf("%s-%s-%d", o.PostageLabel, upNodeName, rnds[0].Int())
+
+	file := bee.NewRandomFile(rnds[0], fileName, o.FileSize)
+
+	batchID, err := upClient.GetOrCreateBatch(ctx, o.PostageAmount, o.PostageDepth, postagelabel)
+	if err != nil {
+		return fmt.Errorf("created batched id %w", err)
+	}
+
+	// upload act file
+	uErr := upClient.UploadActFile(ctx, &file, api.UploadOptions{BatchID: batchID})
+	if uErr != nil {
+		return fmt.Errorf("node %s: %w", upNodeName, uErr)
+	}
+	c.logger.Info("ACT file uploaded")
+
+	time.Sleep(3 * time.Second)
+
+	act := true
+	history := file.HistroryAddress()
+	timestamp := uint64(2)
+
+	// download act file
+	size, hash, err := upClient.DownloadActFile(ctx, file.Address(), &api.DownloadOptions{Act: &act, ActPublicKey: &publisher, ActHistoryAddress: &history, ActTimestamp: &timestamp})
+	if err != nil {
+		return fmt.Errorf("node %s: %w", upNodeName, err)
+	}
+
+	if !bytes.Equal(file.Hash(), hash) {
+		c.logger.Infof("Node %s. ACT file hash not equal. Uploaded size: %d Downloaded size: %d  File: %s", upNodeName, file.Size(), size, file.Address().String())
+		return errors.New("ACT file retrieval - hash error")
+	}
+
+	c.logger.Info("ACT file downloaded")
+
+	// add grantees list
+	gFile := bee.NewBufferFile("grantees.json", bytes.NewBuffer([]byte(`{ "grantees": [
+        "02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e8",
+        "02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12e9",
+        "02ceff1422a7026ba54ad89967d81f2805a55eb3d05f64eb5c49ea6024212b12ee"
+    	]
+	}`)))
+
+	err = upClient.AddActGrantees(ctx, &gFile, api.UploadOptions{BatchID: batchID})
+	if err != nil {
+		return fmt.Errorf("node %s: add grantees error: %w", upNodeName, err)
+	}
+
+	c.logger.Info("ACT grantees added")
+
+	time.Sleep(3 * time.Second)
```

Comment:
Why there is a sleep of 3 seconds, maybe some retry logic is needed as well?

Reply:
without this GetActGrantees return 404